app/services/dataworks_mappers/sdr.rb

```diff
@@ -0,0 +1,471 @@
+        creators:,
+        titles:,
+        publisher:,
+        publication_year:,
+        subjects:,
+        contributors:,
+        descriptions:,
+        dates:,
+        language:,
+        version:,
+        identifiers:,
+        related_identifiers:,
+        sizes:,
+        formats:,
+        rights_list:,
+        funding_references:,
+        related_items:,
+        url:,
+        access:,
+        provider: 'SDR'
+      }.compact_blank
+    end
+    # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
+
+    private
+
+    # Not mapping alternative/subtitles at the moment
+    def titles(element = source.description.title)
+      return [] if element.blank?
+
+      [
+        { title: Cocina::Models::Builders::TitleBuilder.full_title(element).first }
+      ]
+    end
+
+    def identifiers
+      # Always present
+      ids = [
+        { identifier: source.identification.sourceId, identifier_type: 'SourceId' },
+        { identifier: source.externalIdentifier, identifier_type: 'DRUID' }
+      ]
+
+      # May not have a DOI
+      ids.push(doi) if doi.present?
+
+      ids
+    end
+
+    # DOI can be stored in two places, so check both
+    def doi
+      # If it's here, it is not in URL form
+      if source.identification.doi.present?
+        return {
+          identifier: source.identification.doi,
+          identifier_type: 'DOI'
+        }
+      end
+
+      return if (doi = source.description.identifier.find { |id| id.type == 'DOI' }).nil?
+
+      # If it's here, it is in URL form
+      {
+        identifier: id_from_url(doi.value),
+        identifier_type: 'DOI'
+      }
+    end
+
+    # Strip the URL prefix from a DOI (or other identifier) and return just the ID
+    def id_from_url(id_url)
+      URI(id_url).path.delete_prefix('/')
+    end
+
+    def creators
+      source.description.contributor.map do |c|
+        cocina_contributor_struct(c) if c.role.any? { |r| creator_roles.include? r.value.downcase }
+      end.compact
+    end
+
+    def contributors
+      source.description.contributor.map do |c|
+        cocina_contributor_struct(c) unless c.role.any? { |r| creator_roles.include? r.value.downcase }
+      end.compact
+    end
+
+    def publisher
+      source.description.event.flat_map(&:contributor).map do |c|
+        { name: c.name.first.value } if c.role.any? { |r| publisher_roles.include? r.value.downcase }
+      end
+    end
+
+    # Use the earliest year from a publication date, or failing that, the
+    # earliest year from a creation date. If no dates at all, fall back to the
+    # year the source metadata was created.
+    def publication_year
+      first_pub_year = date_values_by_type('Issued').map { |d| Date.edtf(d)&.year }.compact.min
+      created_year = date_values_by_type('Created').map { |d| Date.edtf(d)&.year }.compact.min
+
+      (first_pub_year || created_year || source.created.year).to_s
+    end
+
+    def language
+      source.description.language.pick(:code)
```

Comment:
`&.code`?

Reply:
Isn't `pick` like `.map(:code).first`? I want the first language code, since language is multivalued.